easycv/toolkit/blade/cv_blade_utils.py

```diff
@@ -0,0 +1,248 @@
+    from torch_blade import optimize
+    import ctypes
+    _cudart = ctypes.CDLL('libcudart.so')
+except:
+    pass
+
+
+def blade_env_assert():
+    env_flag = True
+
+    try:
+        import torch
+        torch_version = torch.__version__
+        torch_cuda = torch.version.cuda
+    except:
+        torch_version = 'failed'
+        torch_cuda = 'failed'
+        env_flag = False
+        logging.error(
+            'import torch and torch cuda failed, please install pytorch with cuda correctly'
+        )
+
+    try:
+        import torch_blade
+    except:
+        env_flag = False
+        logging.error(
+            'Import torch_blade failed, please reference to https://help.aliyun.com/document_detail/205134.html'
+        )
+        logging.info('Info: your torch version is %s, your torch cuda version is %s' %
+              (torch_version, torch_cuda))
+
+    try:
+        import torch_blade.tensorrt
+    except:
+        env_flag = False
+        logging.error(
+            'Import torch_blade.tensorrt failed, Install torch_blade.tensorrt and export  xx/tensorrt.so to your python ENV'
+        )
+
+    logging.info('Welcome to use torch_blade, with torch %s, cuda %s, blade %s' %
+          (torch_version, torch_cuda, torch_blade.version.__version__))
+
+    return env_flag
+
+
+@contextmanager
+# def opt_trt_config(enable_fp16=True):
+def opt_trt_config(input_config=dict(enable_fp16=True)): 
+    from torch_blade import tensorrt
+    torch_config = torch_blade.Config()
+    
+    BLADE_CONFIG_KEYS=['optimization_pipeline', 'enable_fp16','customize_op_black_list', 'fp16_fallback_op_ratio']
+
+    for key in input_config.keys():
+        if key in BLADE_CONFIG_KEYS:
+            setattr(torch_config, key, input_config[key])
+            logging.info('setting blade torch_config %s to %s'%(key, input_config[key]))
+
+    torch_config.optimization_pipeline = 'TensorRT'
+    torch_config.enable_fp16 = True
+    torch_config.customize_op_black_list = [
+        'aten::select', 'aten::index', 'aten::slice', 'aten::view'
+    ]
+    torch_config.fp16_fallback_op_ratio = 0.3
+    try:
+        with torch_config:
+            yield
+    finally:
+        pass
+
+
+def cu_prof_start():
+    ret = _cudart.cudaProfilerStart()
+    if ret != 0:
+        raise Exception('cudaProfilerStart() returned %d' % ret)
+
+
+def cu_prof_stop():
+    ret = _cudart.cudaProfilerStop()
+    if ret != 0:
+        raise Exception('cudaProfilerStop() returned %d' % ret)
+
+
+@contextmanager
+def opt_blade_mixprec():
+    try:
+        dummy = torch.classes.torch_blade.MixPrecision(True)
+        yield
+    finally:
+        pass
+
+
+@contextmanager
+def opt_disc_config(enable_fp16=True):
+    torch_config = torch_blade.config.Config()
+    torch_config.enable_fp16 = enable_fp16
+    try:
+        with torch_config:
+            yield
+    finally:
+        pass
+
+
+results = []
+
+
+def printStats(backend, timings, batch_size=1, model_name='default'):
```

Comment:
It seems that the following codes have no relation to Blade but are related to the benchmark. I prefer to add a benchmark tutorial in some place.